ntbs-service-unit-tests/DataAccess/DataQualityRepositoryTests.cs

```diff
@@ -72,6 +62,148 @@ public async Task FiltersOutNotificationsWithTheAlertTypeAlreadyPresent()
+            NotificationStatus status1, NotificationStatus status2, string groupName1, string groupName2,
+            string nhsNumber1, string nhsNumber2, string givenName1, string givenName2, string familyName1,
+            string familyName2, string dob1, string dob2)
+        {
+            // Arrange
+            var group1 = new NotificationGroup();
+            var group2 = new NotificationGroup();
+            await _context.NotificationGroup.AddRangeAsync(group1, group2);
+
+            var notification1 = new Notification
+            {
+                NotificationStatus = status1,
+                NotificationDate = DateTime.Now,
+                Group = groupName1 == "1" ? group1 : groupName1 == "2" ? group2 : null,
+                PatientDetails = new PatientDetails
+                {
+                    NhsNumber = nhsNumber1,
+                    GivenName = givenName1,
+                    FamilyName = familyName1,
+                    Dob = dob1 == null ? null : (DateTime?)DateTime.Parse(dob1)
+                }
+            };
+            var notification2 = new Notification
+            {
+                NotificationStatus = status2,
+                NotificationDate = DateTime.Now,
+                Group = groupName2 == "1" ? group1 : groupName2 == "2" ? group2 : null,
+                PatientDetails = new PatientDetails
+                {
+                    NhsNumber = nhsNumber2,
+                    GivenName = givenName2,
+                    FamilyName = familyName2,
+                    Dob = dob1 == null ? null : (DateTime?)DateTime.Parse(dob2)
+                }
+            };
+
+            await _context.AddRangeAsync(notification1, notification2);
+            await _context.SaveChangesAsync();
+
+            var repo = new DataQualityRepository(_context);
+
+            try
+            {
+                // Act
+                var notificationIds = await repo.GetNotificationIdsEligibleForDqPotentialDuplicateAlertsAsync();
+
+                // Assert
+                if (duplicateShouldBeFound)
+                {
+                    Assert.Contains(notificationIds, pair =>
+                        pair.NotificationId == notification1.NotificationId
+                        && pair.DuplicateId == notification2.NotificationId);
+                    Assert.Contains(notificationIds, pair =>
+                        pair.NotificationId == notification2.NotificationId
+                        && pair.DuplicateId == notification1.NotificationId);
+                    Assert.Equal(2, notificationIds.Count);
+                }
+                else
+                {
+                    Assert.Empty(notificationIds);
+                }
+            }
+            finally
+            {
+                _context.RemoveRange(group1, group2, notification1, notification2);
+                await _context.SaveChangesAsync();
+            }
```

Comment:
An alternative here is to tear down the tests in a `Dispose` method (as well as having the `Dispose` on the fixture that you already have). Semantically this would be a bit nicer, but it would have the downside of requiring the tests to track which groups/notifications they've added in a class field, rather than just as local variables.

I'm not saying you should change it, I just want to make sure you're aware of the different options.

Reply:
Yep I had considered this, but had decided against this for exactly that reason. However, having thought about it a bit more, I can just empty the entire database at the end of each test, which I've now done in `Dispose`